### ISSUE TITLE
(XMB) Add ability to use custom shader for bakcground

### ISSUE THREAD

**Issue by no-boot-device** [libretro/RetroArch#9242]
It has all these perfectly nice shaders in the `procedural` folder and yet the list of shaders you can use is baked in. I figure there might be performance issues maybe, but I don't see why it can't be an option?

**Comment by hizzlekizzle** [comment]
Yes, this would be great, but it's unfortunately more difficult to implement in the menu code than it might seem.

The snow and bokeh shaders draw a fullscreen quad, which is what's needed for these kinds of effects, so you can use them as a base for baking your own custom shaders into the menu, but it's a hassle, for sure. It would be much easier to have a 'custom' option in the shader pipeline menu where it would just load up an arbitrary file...

Vulkan would be a problem, though, as it requires offline compiling of shaders to spir-v, IIRC.

**Comment by bparker06** [comment]
How do slang shaders in-game currently work if vulkan requires offline compiling?

**Comment by hizzlekizzle** [comment]
it only requires it for the menu and for beetle-psx. I'm not sure why the difference, to be honest. for beetle-psx, I'm assuming for performance and to prevent stuttering from having to compile them on-the-fly, but that wouldn't be an issue for the menu, anyway :man_shrugging:
